url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/StandardBeanInfo.java

```diff
@@ -0,0 +1,1505 @@
+        paramTypes = theMethod.getParameterTypes();
+        if (paramTypes.length > 1
+                || (paramTypes.length == 1 && paramTypes[0] != int.class)) {
+            return;
+        }
+
+        table = propertyTable.get(propertyName);
+        if (table == null) {
+            table = new HashMap();
+            propertyTable.put(propertyName, table);
+        }
+
+        getters = (ArrayList<Method>) table.get(STR_GETTERS);
+        if (getters == null) {
+            getters = new ArrayList<Method>();
+            table.put(STR_GETTERS, getters);
+        }
+
+        // add current method as a valid getter
+        getters.add(theMethod);
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    private static void introspectSet(Method theMethod,
+            HashMap<String, HashMap> propertyTable) {
+
+        String methodName = theMethod.getName();
+        if (methodName == null) {
+            return;
+        }
+        String propertyName;
+        Class returnType;
+        Class[] paramTypes;
+
+        // setter method should never return type other than void
+        returnType = theMethod.getReturnType();
+        if (returnType != void.class) {
+            return;
+        }
+
+        if (methodName == null || !methodName.startsWith(PREFIX_SET)) {
+            return;
+        }
+
+        propertyName = decapitalize(methodName.substring(PREFIX_SET.length()));
+
+        // validate property name
+        if (!isValidProperty(propertyName)) {
+            return;
+        }
+
+        // It seems we do not need to validate return type
+
+        // validate param types
+        paramTypes = theMethod.getParameterTypes();
+
+        if (paramTypes.length == 0 || paramTypes.length > 2
+                || (paramTypes.length == 2 && paramTypes[0] != int.class)) {
+            return;
+        }
+
+        HashMap table = propertyTable.get(propertyName);
+        if (table == null) {
+            table = new HashMap();
+            propertyTable.put(propertyName, table);
+        }
+
+        ArrayList<Method> setters = (ArrayList<Method>) table.get(STR_SETTERS);
+        if (setters == null) {
+            setters = new ArrayList<Method>();
+            table.put(STR_SETTERS, setters);
+        }
+
+        // handle constrained
+        Class[] exceptions = theMethod.getExceptionTypes();
+        for (Class e : exceptions) {
+            if (e.equals(PropertyVetoException.class)) {
+                table.put(STR_IS_CONSTRAINED, Boolean.TRUE); //$NON-NLS-1$
+            }
+        }
+
+        // add new setter
+        setters.add(theMethod);
+    }
+
+    /**
+     * Checks and fixs all cases when several incompatible checkers / getters
+     * were specified for single property.
+     * 
+     * @param propertyTable
+     * @throws IntrospectionException
+     */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    private void fixGetSet(HashMap<String, HashMap> propertyTable)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 95. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.